src/pdf.py

```diff
@@ -200,7 +200,7 @@ def __init__(self, doc: 'pml.Document'):
         self.doc: pml.Document = doc
 
     # generate PDF document and return it as a string
-    def generate(self) -> str:
+    def generate(self) -> AnyStr:
```

Comment:
```suggestion
    def generate(self) -> bytes:
```

The [`str.encode` method](https://docs.python.org/3/library/stdtypes.html#str.encode), which is used by the underlying call to `util.toLatin1`, returns `bytes` rather than `AnyStr`. There's a lot of history involved in the usage of `str` and `bytes` across Python 2 and 3, but unless this project needs to support Python 2, we should most likely be using `str` and `bytes` where appropriate rather than `AnyStr` (this is a bit of a generalized statement).

Is it a general goal of this project to adopt type annotations? If so, I'd be interested in filing an issue to use `mypy` for static type checking, since declared but unchecked types may lead to subtle bugs and confusion for new developers.

Reply:
Thanks!

I am a huge fan of static type analysis. If we had it, it would have prevented me from breaking the PDF preview in the first place. If `mypy` can be configured so that it checks the given types, but doesn't make problems for the code that doesn't have type annotations yet, I think it would be great if Trelby used it. @limburgher What do you think?